### PR TITLE
Propagate all experimental options to the exec configuration

### PR DESCRIPTION
**Pull request by fmeum**
Experimental flags that don't propagate to the exec configuration can create very surprising failure cases for both Bazel users and devs. Unless there is a valid reason to not propagate them, they should be. 

This is now enforced by a test and existing flags have been cleaned up where possible. The test strategy is the same as for `incompatible_` flags.

### CODE REVIEW

**Comment by fmeum** [comment]
This is motivated by https://github.com/bazelbuild/bazel/pull/23328.

**Comment by fmeum** [comment]
@lberki Could you review this as Bazel's "Chief Flag Officer"? :⁠-⁠)
@aranguyen for configurability

**Comment by fmeum** [comment]
@lberki Friendly ping

**Comment by lberki** [comment]
Uh, sorry. I'll defer to @gregestren here. This seems like a good idea on the surface. You also probably want to give `--incompatible_*` the same treatment.

**Comment by fmeum** [comment]
I added a test for incompatible tests about a year ago, but back then I wasn't sure whether experimental flags should be treated in the same way.

File: src/test/java/com/google/devtools/build/lib/analysis/config/ExecutionTransitionFactoryTest.java [hunk 1]
```diff
@@ -279,10 +279,80 @@ public void incompatibleOptionsPreservedInExec() throws Exception {
...
+                        || o.getValue().hasOptionMetadataTag(OptionMetadataTag.EXPERIMENTAL))
+            .filter(o -> o.getValue().getDefinition().getType().isAssignableFrom(boolean.class))
+            .filter(o -> !o.getValue().getDefinition().isDeprecated())
+            .filter(
+                // Skipping this explicitly as propagating it causes a cycle when compiling the
+                // optimizer itself.
+                o -> !o.getKey().equals("experimental_local_java_optimizations"))
+            .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
+
+    // Verify all "--experimental_*" options also have the INCOMPATIBLE_CHANGE metadata tag.
```
Comment (gregestren) [inline src/test/java/com/google/devtools/build/lib/analysis/config/ExecutionTransitionFactoryTest.java:309]: Comment typo?

Comment (fmeum) [reply src/test/java/com/google/devtools/build/lib/analysis/config/ExecutionTransitionFactoryTest.java]: Yes, fixed

**Review by gregestren — APPROVED** [review]

**Comment by gregestren** [comment]
Do we need re-review?

**Comment by fmeum** [comment]
> Do we need re-review?

Sorry, clicked the button accidentally while going over my pending PRs.

File: src/main/starlark/builtins_bzl/common/builtin_exec_platforms.bzl [hunk 2]
```diff
@@ -130,6 +130,28 @@ bazel_fragments["AndroidConfiguration.Options"] = fragment(
         "//command_line_option:internal_persistent_busybox_tools",
         "//command_line_option:internal_persistent_multiplex_busybox_tools",
         "//command_line_option:incompatible_disable_native_android_rules",
+        "//command_line_option:android_databinding_use_androidx",
```
Comment (gregestren) [inline src/main/starlark/builtins_bzl/common/builtin_exec_platforms.bzl:133]: Hi @fmeum . We're trying to get this merged for Bazel 8.

Why were these flags added? Does this have something to do with the `experimental_android_databinding_v2`below?

Comment (fmeum) [reply src/main/starlark/builtins_bzl/common/builtin_exec_platforms.bzl:133]: It's possible that they have been deleted in the meantime and I just didn't notice during the rebase. I didn't mean to add new flags, I just wanted to propagate all existing experimental flags.

**Comment by hvadehra** [comment]
Just a heads up, this is being rolled back, it breaks things internally. But the number of breakages are very small, so it's probably safe to keep for Bazel.

**Comment by gregestren** [comment]
This is still locked in for Bazel 8, right? 

I'll look at getting it rolled forward again at head. Yes, the # of breakages is small. And they're almost certainly due to properties of the Google depot which don't generalize to Bazel, so there's no reason to block this outside that specific depot.

**Pull request closed without merging**